doc/specs/Keyboard-Selection.md

```diff
@@ -58,21 +61,48 @@ Every combination of the `Direction` and `SelectionExpansionMode` will map to a
+
+#### `toggleMarkMode`
+`toggleMarkMode` has two keybinding arguments:
+| Parameter | Accepted Values | Default Value | Description |
+|--|--|--|--|
+| `anchorKeybinding` | regular keybindings and plain modifier keys | `Shift` | The keybinding used to anchor the selection anchor to it's current text buffer position. While holding down that keybinding, any `moveSelectionAnchor` actions will move the other selection anchor (`endSelectionPosition`). |
+| `anchorMode` | `Hold`, `Toggle` | `Hold` | If `Hold`, the user needs to hold the `anchorKeybinding` keybinding to move the second selection anchor. If `Toggle`, the user switches between selection anchors when they press it. |
+
+#### Rendering during Mark Mode
+Since we are just moving the selection anchors, rendering the selection rects should operate normally. We need to ensure that we still scroll when we move a selection anchor past the top of bottom of the viewport.
+
+In ConHost, output would be paused when a selection was present. This is a completely separate issue that is being tracked in (#2529)[https://github.com/microsoft/terminal/pull/2529].
+
+#### Exiting Mark Mode
+There are multiple ways to exit mark mode. The user can press...
+- the `toggleMarkMode` keybinding
+- the `ESC` key
+- the `copy` keybinding (which also copies the contents of the selection to the user's clipboard)
+
+
+
+<!--Holding down the `holdSelectionAnchor` keybinding action (set to `Shift` by default) anchors the selection anchor to it's current text buffer position. While holding down that keybinding, any `moveSelectionAnchor` actions will move the other selection anchor (`endSelectionPosition`).
+
+**NOTE**: `holdSelectionAnchor` will have an `Enum anchorSwitching` parameter. Where `Hold` (default) is the functionality mentioned above, and `Toggle` switches between the selection anchors when pressed.-->
 
 
 ## UI/UX Design
 
 ### Keybindings
 
-Thanks to Keybinding Args, there will only be 3 new commands that need to be added:
+Thanks to Keybinding Args, there will only be 4 new commands that need to be added:
 | Action | Keybinding Args | Description |
 |--|--|--|
 | `moveSelectionAnchor` |                                                               | If a selection exists, moves the last selection anchor.
 |                       | `Enum direction { up, down, left, right}`                     | The direction the selection will be moved in. |
 |                       | `Enum expansionMode { cell, word, line, viewport, buffer }`   | The context for which to move the selection anchor to. (defaults to `cell`)
 | `selectEntireBuffer`  | | Select the entire text buffer.
 | `toggleMarkMode`      | | Enter or exit mark mode. This allows you to create an entire selection using only the keyboard. |
+| `holdSelectionAnchor` | | Enter or exit mark mode. This allows you to create an entire selection using only the keyboard. |

```

Comment:
Is this intended to be removed? Or is the text above supposed to be uncommented?

Reply:
Idk what I was thinking. This was a long time ago haha. Removed.

Also, mark mode discussion just got moved to #5804 to make our lives easier.